subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -449,6 +718,10 @@ SHELL_STATIC_SUBCMD_SET_CREATE(wifi_commands,
 	SHELL_CMD(scan, NULL, "Scan for Wi-Fi APs", cmd_wifi_scan),
 	SHELL_CMD(status, NULL, "Status of the Wi-Fi interface", cmd_wifi_status),
 	SHELL_CMD(statistics, NULL, "Wi-Fi interface statistics", cmd_wifi_stats),
+	SHELL_CMD(ps, NULL, "Configure Wi-Fi power save on/off, no arguments will dump config",
+		  cmd_wifi_ps),
```

Comment:
Is there a reason to have ps_mode as a separate command? Would it be better to combine it with the ps command i.e. force the user to pass the ps_mode if he is setting ps to on?

Reply:
Yes, `ps_mode` is a one-time configuration (esp. for WMM), where as `ps` can be done multiple-times and most users may not care about `ps_mode` (some default selected by chip should be good enough), but they want to control `ps`, so, separated the commands.